dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java

```diff
@@ -264,6 +242,52 @@ public static double getJavaCpuUsage() {
      * @return double system cpu usage (returns -1 if info not available)
      */
     public static double getSystemCpuUsage() {
-        return instance().getSystemCPU();
+        return INSTANCE.getSystemCPU();
+    }
+
+    public static MemoryInfo createMemoryInfo() {
+        java.lang.management.OperatingSystemMXBean mbean = ManagementFactory.getOperatingSystemMXBean();
+
+        if (JavaInfo.vendor() == Vendor.IBM) {
+            return new IBMJavaMemoryInfo(mbean);
+        } else {
+            return new ModernJavaMemoryInfo(mbean);
+        }
+
+    }
+
+    private static interface MemoryInfo {
```

Comment:
MemoryInfo is a weird name for this interface and it implementors.  There is nothing about Memory in the classes.  I would think something like SystemCpuInfo for the interface and just [IBMJava|ModernJava]CpuInfo for the implementations.